server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -939,7 +939,8 @@ public Query termQuery(Object value, QueryShardContext context) {
 
         @Override
         protected void parseCreateField(ParseContext context) throws IOException {
-            //field defined as runtime field, don't index anything
+            //field defined as runtime field, don't index anything but skip over any possible children
+            context.parser().skipChildren();
```

Comment:
I am not 100% sure that this is the expected behaviour. The way they are applied, runtime fields are an additional layer on top of ordinary fields. because an object field is shadowed by a runtime field, we end up skipping all of the children structure and not index fields that would not be directly shadowed. If doable, could we consider instead mapping objects as objects even if they are shadowed? That would be more inline with the statement that "objects cannot be defined in the runtime section".